extension/storage/nop_client.go

```diff
@@ -0,0 +1,40 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package storage // import "go.opentelemetry.io/collector/extension/storage"
+
+import "context"
+
+type nopClient struct{}
+
+var nopClientInstance Client = &nopClient{}
+
+// NewNopClient returns a nop client
+func NewNopClient() Client {
```

Comment:
In general we have these no-op in a ...test package, should we do the same?

Reply:
This is not intended specifically as a test component. Previous discussion on the purpose:
- https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/2883#discussion_r606296192
- https://github.com/open-telemetry/opentelemetry-collector/pull/8379#discussion_r1320537822